src/test/java/com/puppycrawl/tools/checkstyle/checks/NewlineAtEndOfFileCheckTest.java

```diff
@@ -110,8 +110,9 @@ public void testNoNewlineLfAtEndOfFile() throws Exception {
 
     @Test
     public void testNoNewlineAtEndOfFile() throws Exception {
+        final String msgKeyNoNewlineEof = "File does not end with a newline :)";
```

Comment:
This should cause issues with localization because this now means this test can only pass on an english machine if we do not enforce a locale during the run.

Reply:
I can annotate it with `@DefaultLocale` to enforce the locale as after the annotated element has been executed the initial default value is restored.

There are similar pieces of code already present: https://github.com/checkstyle/checkstyle/blob/74d8802acfb8833cd34538447d0852baa58e6c5b/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheckTest.java#L112-L120

Can't a machine in a different locale understand strings in different languages? Like I can write a string in Chinese on my locale. I am not sure how failure will occur. 

I executed the test with `@DefaultLocale("zh-Hant-TW")` and `@DefaultLocale(language = "fr")`, both the times the test passed.

Reply:
Locale testing may not fully work until https://github.com/checkstyle/checkstyle/pull/11529 .

> Can't a machine in a different locale understand strings in different languages? 

The issue is your assertion is only looking for english, while the check execution may use the current locale and we pull in messages from different message files based on the locale requested. If we pull in french messages and compare then to your english message, the assertion will fail even though there is no "issue".

> There are similar pieces of code already present

`InputMatchXpathDoubleBrace.java` uses a custom message that is hardcoded in english in the configuration. This does not count.

Reply:
I guess I missed this, but I see your using a hardcoded messsage in the configuration for your tests.

Why do we need a hardcoded message here?